opm/simulators/wells/MultisegmentWellSegments.hpp

```diff
@@ -74,15 +74,18 @@ class MultisegmentWellSegments
     EvalWell getFrictionPressureLoss(const int seg, const bool return_upwind_derivatives) const;
 
     // pressure drop for Spiral ICD segment (WSEGSICD)
-    EvalWell pressureDropSpiralICD(const int seg) const;
+    EvalWell pressureDropSpiralICD(const int seg,
+                                   const bool return_extra_derivatives) const;
 
     // pressure drop for Autonomous ICD segment (WSEGAICD)
     EvalWell pressureDropAutoICD(const int seg,
-                                 const UnitSystem& unit_system) const;
+                                 const UnitSystem& unit_system,
+                                 const bool return_extra_derivatives) const;
 
     // pressure drop for sub-critical valve (WSEGVALV)
-    EvalWell pressureDropValve(const int seg) const;
-
+    EvalWell pressureDropValve(const int seg, 
+                               const bool return_extra_derivatives) const;
```

Comment:
Maybe we could have a default value of `const bool return_extra_derivatives = false` for these functions?  That way the existing calls could remain unchanged while the new function calls for the "extra" derivatives would stand out more.